apps/notifications/src/alert/services/deployment-balance-alerts/deployment-balance-alerts.service.ts

```diff
@@ -0,0 +1,108 @@
+import { Injectable } from '@nestjs/common';
+import template from 'lodash/template';
+
+import { ChainBlockCreatedDto } from '@src/alert/dto/chain-block-created.dto';
+import {
+  DeploymentBalanceAlertOutput,
+  DeploymentBalanceAlertRepository,
+} from '@src/alert/repositories/deployment-balance-alert/deployment-balance-alert.repository';
+import { ConditionsMatcherService } from '@src/alert/services/conditions-matcher/conditions-matcher.service';
+import { DeploymentService } from '@src/alert/services/deployment/deployment.service';
+import { BrokerService } from '@src/broker';
+import { LoggerService } from '@src/common/services/logger.service';
+
+type AlertCallback = (
+  alert: DeploymentBalanceAlertOutput,
+) => Promise<void> | void;
+
+@Injectable()
+export class DeploymentBalanceAlertsService {
+  constructor(
+    private readonly alertRepository: DeploymentBalanceAlertRepository,
+    private readonly conditionsMatcher: ConditionsMatcherService,
+    private readonly brokerService: BrokerService,
+    private readonly deploymentService: DeploymentService,
+    private readonly loggerService: LoggerService,
+  ) {
+    this.loggerService.setContext(DeploymentBalanceAlertsService.name);
+  }
+
+  private readonly DEPLOYMENT_BALANCE_BLOCKS_THROTTLE = 10;
+
+  async alertFor(block: ChainBlockCreatedDto): Promise<void> {
+    await this.forEachAlert(block.height, async (alert) =>
+      this.processSingleAlert(block, alert),
+    );
+  }
+
+  private async forEachAlert(block: number, callback: AlertCallback) {
+    try {
+      await this.alertRepository.paginate({
+        query: { block },
+        limit: 10,
+        callback: async (alerts: DeploymentBalanceAlertOutput[]) => {
+          await Promise.all(alerts.map(async (alert) => callback(alert)));
+        },
+      });
+    } catch (error) {
+      this.loggerService.error({
+        event: 'ALERT_FAILURE',
+        block,
+        error,
+      });
+      throw error;
+    }
+  }
+
+  private async processSingleAlert(
+    block: ChainBlockCreatedDto,
+    alert: DeploymentBalanceAlertOutput,
+  ) {
+    try {
+      const balance = await this.deploymentService.getDeploymentBalance(
+        alert.owner,
+        alert.dseq,
+      );
+
+      if (!balance) {
```

Comment:
thought(non-blocking): `balance` may be `0` and if so, then it will disable alert but probably should not